src/renderer/features/network/ManageCustomRpcNode/model/add-custom-rpc-model.ts

```diff
@@ -0,0 +1,178 @@
+import { createEffect, createEvent, createStore, sample } from 'effector';
+import { createForm } from 'effector-forms';
+
+import { networkService } from '@shared/api/network';
+import { ExtendedChain } from '@entities/network';
+import { CheckRpcNodeFxParams, NodeExistParam, RpcCheckResult, SaveRpcNodeFxParams } from '../lib/types';
+import { manageNetworkModel } from '@pages/Settings/Networks/model/manage-network-model';
+import { fieldRules, RpcValidationMapping } from '../lib/constants';
+
+const $addCustomRpcForm = createForm({
+  fields: {
+    name: {
+      init: '',
+      rules: fieldRules.name,
+    },
+    url: {
+      init: '',
+      rules: fieldRules.url,
+    },
+  },
+  validateOn: ['change'],
+});
+
+const formInitiated = createEvent();
+const networkChanged = createEvent<ExtendedChain>();
+const rpcConnectivityChecked = createEvent<RpcCheckResult>();
+const nodeExistChecked = createEvent<boolean>();
+const processStarted = createEvent<boolean>();
+
+const $selectedNetwork = createStore<ExtendedChain | null>(null);
+const $rpcConnectivityResult = createStore<RpcCheckResult>(RpcCheckResult.INIT);
+const $isNodeExist = createStore<boolean>(false);
+const $isProcessStarted = createStore<boolean>(false);
+const $isLoading = createStore<boolean>(false);
+
+const checkRpcNodeFx = createEffect(async ({ network, url }: CheckRpcNodeFxParams): Promise<RpcCheckResult> => {
+  const validationResult = await networkService.validateRpcNode(network.chainId, url);
+  const result = RpcValidationMapping[validationResult];
+
+  return result;
+});
+
+const saveRpcNodeFx = createEffect(async ({ network, form }: SaveRpcNodeFxParams) => {
+  manageNetworkModel.events.rpcNodeAdded({
+    chainId: network?.chainId,
+    rpcNode: {
+      name: form.name,
+      url: form.url,
+    },
+  });
+});
+
+const isNodeExistFx = createEffect(({ network, url }: NodeExistParam): boolean => {
```

Comment:
`effect` is an action so verbs must be used in the first place
Here `check` is a nice candidate - `checkIsNodeExistFx`

Reply:
By the way this `effect` doesn't involve any kind of side effect
Every part of this code could be placed in `fn` for appropriate `sample`